File: 0782-jewels-and-stones/0782-jewels-and-stones.py
class Solution:
    def numJewelsInStones(self, jewels: str, stones: str) -> int:
        # j_set = set(jewels)
        # count = 0
        # for stone in stones:
        #     if stone in j_set:
        #         count += 1
        
        # return count

        return sum(1 for stone in stones if stone in set(jewels))

File: 0782-jewels-and-stones/0782-jewels-and-stones.go
func numJewelsInStones(jewels string, stones string) int {
    jewelSet := make(map[rune]struct{})

    for _, jewel := range jewels {
        jewelSet[jewel] = struct{}{}
    }

    count := 0

    for _, stone := range stones {
        if _, ok := jewelSet[stone]; ok {
            count++
        }
    }

    return count
}